nsis/src/nsis/NsisLoader.java

```diff
@@ -176,12 +176,14 @@ private void createGhidraMemoryBlock(InputStream is, Address startingAddr, Progr
 			boolean writePermission, boolean executePermission)
 			throws LockException, MemoryConflictException, AddressOverflowException,
 			CancelledException, DuplicateNameException {
-		Memory memory = program.getMemory();
-		MemoryBlock firstHeaderBlock = memory.createInitializedBlock(blockName, startingAddr, is,
-				size, monitor, /* Overlay */ false);
-		firstHeaderBlock.setRead(readPermission);
-		firstHeaderBlock.setWrite(writePermission);
-		firstHeaderBlock.setExecute(executePermission);
+		if (size > 0) {
```

Comment:
When `createGhidraMemoryBlock` is called I think it is always followed by `createData` should that call be skipped if size is not greater than 0?

Reply:
oh yes, good catch.

Then should I add the condition in each init function?

Reply:
Yes, I think that's correct.

What do you think about creating a test case for this?

To create a test case, you would make a copy of the no compression test case and manually modify one or more bytes to create a 0 size section to act like the sample that failed. Do you understand how you would do this?

Reply:
yes that makes sense to me

I'm wondering if the test case is worth, because I already check the number of objects in each block header (ie `ne.getNumPages()`). When the memory block size is 0, it just means that I got 0 as the number of objects in one block header. With a test case, I won't be testing the code in NsisLoader

Reply:
Ya, I see your point. In this case the failure seemed to be in the ghidra code, which don't really have a good way to test right now. I don't think a test case is needed in this case.

Just keep in mind how you would do it in the future.